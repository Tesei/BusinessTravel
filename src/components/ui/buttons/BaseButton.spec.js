import { describe, it, expect } from 'vitest'
import { mount } from '@vue/test-utils'
import component from './BaseButton.vue'

describe('BaseButton component', () => {
    const buildWrapper = (disabled = false, toShowLoading = false) => {
        const propsData = {
            disabled,
            classNames: 'test-class',
            toShowLoading,
        }
        return mount(component, {
            propsData,
            slots: {
                default: 'Окно ввода',
            },
        })
    }
    it('mounts without error', () => {
        const wrapper = buildWrapper()
        expect(wrapper.exists).toBeTruthy()
        expect(wrapper.html()).toMatchSnapshot()
    })
    it('don have data - disabled button', () => {
        const wrapper = buildWrapper(true)
        expect(wrapper.find('button').attributes().disabled).toBe('')
    })
    it('if we have all data button - active', () => {
        const wrapper = buildWrapper(false)
        expect(wrapper.find('button').attributes().class).includes('test-class')
    })
    it('button has attribute submit', () => {
        const wrapper = buildWrapper(false)
        expect(wrapper.find('button').attributes().type).includes('submit')
    })
    it('button has attribute submit', () => {
        const wrapper = buildWrapper(false, true)
        expect(wrapper.find('button').text()).not.toContain('Окно ввода')
    })
    it('button slot has text', () => {
        const wrapper = buildWrapper()
        expect(wrapper.find('button').text()).toContain('Окно ввода')
    })
})
