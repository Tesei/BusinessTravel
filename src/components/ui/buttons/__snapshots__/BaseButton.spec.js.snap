// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BaseButton component > mounts without error 1`] = `"<button data-v-170f5cbe="" class="button font-button_M jc-c test-class" type="submit">Окно ввода</button>"`;

exports[`Button component > mounts without error 1`] = `"<button data-v-170f5cbe="" class="button font-button_M jc-c test-class" type="submit">Окно ввода</button>"`;
